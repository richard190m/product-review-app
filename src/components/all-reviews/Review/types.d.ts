export interface ReviewProps {
  data: Review;
}
