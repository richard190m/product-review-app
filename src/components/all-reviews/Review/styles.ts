import { Happy, Meh, Sad } from '@styled-icons/boxicons-regular';
import { Star as StarSolid } from '@styled-icons/boxicons-solid';
import styled from 'styled-components';

export const ReviewStyled = styled.div`
  display: flex;
`;

export const Author = styled.div`
  width: 60px;
  margin: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
`;

export const IconSad = styled(Sad)`
  width: 30px;
`;

export const IconMeh = styled(Meh)`
  width: 30px;
`;

export const IconHappy = styled(Happy)`
  width: 30px;
`;

export const InfoReview = styled.div`
  display: flex;
  flex-direction: column;
  margin: 10px 0;
  width: 100%;
`;

export const StarFilled = styled(StarSolid)`
  width: 30px;
  color: #febd09;
`;

export const StarStyled = styled(StarSolid)`
  width: 30px;
  color: #ededed;
`;

export const CommentArea = styled.div`
  background-color: #dedede;
  padding: 20px;
  border-radius: 10px;
`;

export const Comment = styled.p`
  font-size: 14px;
  font-weight: 300;
`;

export const DateStyled = styled.span`
  font-size: 14px;
  font-weight: 300;
  padding-left: 10px;
  color: gray;
`;

export const Rating = styled.p`
  display: flex;
  align-items: center;
`;
