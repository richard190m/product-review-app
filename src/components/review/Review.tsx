import {
  Author,
  Comment,
  CommentArea,
  DateStyled,
  IconHappy,
  IconMeh,
  IconSad,
  InfoReview,
  Rating,
  ReviewStyled,
  StarFilled,
  StarStyled,
} from './styles';
import React, { useEffect, useState } from 'react';
import formatDate from '../../utils/formatDate';
import PropTypes from 'prop-types';
import { ReviewProps } from './types';

const Review = ({ data }: ReviewProps) => {
  const [ratings, setRatings] = useState(new Array(5).fill(false));
  const { rating, name, title, comment, date } = data;
  const dateFormated = formatDate(date);

  useEffect(() => {
    setRatings(ratings.map((item, index) => index <= rating - 1));
  }, [rating]);

  return (
    <ReviewStyled>
      <Author>
        {rating < 3 ? <IconSad /> : rating > 3 ? <IconHappy /> : <IconMeh />}
        <p>{name}</p>
      </Author>
      <InfoReview>
        <Rating>
          {ratings.map((value, index) =>
            value ? <StarFilled key={index} /> : <StarStyled key={index} />,
          )}
          <DateStyled>{`(${dateFormated})`}</DateStyled>
        </Rating>
        <CommentArea>
          <p>{title}</p>
          <Comment>{comment}</Comment>
        </CommentArea>
      </InfoReview>
    </ReviewStyled>
  );
};

Review.propTypes = {
  data: PropTypes.object.isRequired,
};

export default Review;
