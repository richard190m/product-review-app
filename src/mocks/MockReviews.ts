export default [
  {
    name: 'Name 1',
    email: 'test@test.com',
    rating: 2,
    title: 'Not so good product',
    comment: 'Not recomended...',
    date: new Date(2022, 1, 1),
  },
  {
    name: 'Name 2',
    email: 'test@test.com',
    rating: 3,
    title: 'Not bad, not good',
    comment: "It's a product that can do the job.",
    date: new Date(2022, 3, 1),
  },
  {
    name: 'Name 3',
    email: 'test@test.com',
    rating: 5,
    title: 'Good product',
    comment: 'Best product I ever used',
    date: new Date(2022, 4, 1),
  },
];
