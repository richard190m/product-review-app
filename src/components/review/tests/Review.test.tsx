import MockReviews from '../../../mocks/MockReviews';
import { render } from '@testing-library/react';
import Review from '../Review';

describe('Review', () => {
  test('should render correctly', () => {
    const { container } = render(<Review data={MockReviews[0]} />);

    expect(container).toMatchSnapshot();
  });
});
