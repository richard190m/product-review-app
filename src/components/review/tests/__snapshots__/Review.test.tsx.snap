// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ emMGHK"
  >
    <div
      class="sc-gsnTZi cTauWR"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-ea9ulj-0 lbJwfL sc-dkzDqf gPTPlL"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
        />
        <circle
          cx="8.5"
          cy="10.5"
          r="1.5"
        />
        <circle
          cx="15.493"
          cy="10.493"
          r="1.493"
        />
        <path
          d="M12 14c-3 0-4 3-4 3h8s-1-3-4-3z"
        />
      </svg>
      <p>
        Name 1
      </p>
    </div>
    <div
      class="sc-jSMfEi iOkfAM"
    >
      <p
        class="sc-kDDrLX bhNGxW"
      >
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 lbJwfL sc-gKXOVf iwfGiM"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 lbJwfL sc-gKXOVf iwfGiM"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 lbJwfL sc-iBkjds dAGWXP"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 lbJwfL sc-iBkjds dAGWXP"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 lbJwfL sc-iBkjds dAGWXP"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
          />
        </svg>
        <span
          class="sc-jqUVSM cukEmc"
        >
          (01/02/2022)
        </span>
      </p>
      <div
        class="sc-ftvSup ksgWSR"
      >
        <p>
          Not so good product
        </p>
        <p
          class="sc-papXJ IZftc"
        >
          Not recomended...
        </p>
      </div>
    </div>
  </div>
</div>
`;
